### PR TITLE
Issue #2806011. Avoid calling drupal_goto() directly from rules actions.

### PR DESCRIPTION
**Pull request by xurizaemon**
See https://www.drupal.org/project/civicrm_entity/issues/2806011

### CODE REVIEW

File: modules/civicrm_entity_actions/civicrm_entity_actions_action_schedule.inc [hunk 1]
```diff
@@ -41,26 +41,26 @@ function civicrm_entity_actions_action_schedule_action_info() {
 
 function civicrm_entity_actions_action_schedule_add_action($entity, $context = array()) {
   $base_url = str_replace('_', '-', $context['entity_type']); // civicrm-action_schedule
-  drupal_goto($base_url . '/add');
+  rules_action_drupal_goto($base_url . '/add');
```
Comment (jackrabbithanna) [inline modules/civicrm_entity_actions/civicrm_entity_actions_action_schedule.inc]: If you use this function then this module will need to add a dependency to Rules in the .info file

Comment (jackrabbithanna) [reply modules/civicrm_entity_actions/civicrm_entity_actions_action_schedule.inc]: alternatively you could add a if (module_exists('rules')) { new function, else drupal_goto

**Comment by jackrabbithanna** [comment]
@xurizaemon I'm going to be cutting 7.x-2.0 stable next week. If you have time, and want to finish this off, we can get it in the 2.0 release

**Comment by xurizaemon** [comment]
Sorry I'd missed yr earlier comments. Second approach seems sensible.

IDK if this is really a problem for anyone in the real world, so I'm OK with it not being merged!

I had five minutes to implement your suggestion but I don't have time / available test sites to test this with right now.

**Comment by jackrabbithanna** [comment]
Your solution seems great for this in your second commit.  I'll test this and it'll go into 2.0. Thanks alot :)

**Comment by xurizaemon** [comment]
no-worries-alot

**Comment by jackrabbithanna** [comment]
Merging #129

**Pull request merged**